okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelector.java

```diff
@@ -51,4 +51,9 @@
      * okhttp3.Request#url} does.
      */
     List<HttpUrl> getBaseUrls();
+
+    /**
+     * Marks the specified URL as failed, which removes it from the pool of available URLs.
```

Comment:
"Indicates that a call against the given URL has failed. Implementations can use failure statistics to determine which hosts may be unavailable and should be avoided for future calls."